### ISSUE TITLE
Captive portal without ethernet

### ISSUE THREAD

**Issue by Tywors** [pihomeserver/Kupiki-Hotspot-Script#75]
Hello!
Why don't show the captive portal if the raspberry don't have internet (ethernet cable)?

How I can resolve this, please?

**Comment by pihomeserver** [comment]
Sorry i Don't understand what is your expected behavior. Can you please provide more details

**Comment by Tywors** [comment]
Hi!
I need kupiki only for navegate in Lan websites, for wikis and docs. But if i disconnect the ethernet from the raspberry, i can connect to hotspot but don't show the captive website.

Sorry for my english, 

Cool project! thanks

**Comment by pihomeserver** [comment]
Ok so let me guess why it does not work. If you remove the ethernet link, then the interface is not set with an IP, etc. Then coova rules in up.sh with IP tables will not work at all.
Try to remove added rules in up.sh.

**Comment by Tywors** [comment]
Thanks! I try tomorrow! Where is the up.sh?

**Comment by pihomeserver** [comment]
You will have to remove the added rules at the end of the script
If there is no cable i'm pretty sure that the IP will not be set and then redirection rules will fail

**Comment by Tywors** [comment]
Hi, i'm using your prepare .img
Would you know what number of lines they are?

I'm not able to make it work. Thanks!

**Comment by pihomeserver** [comment]
The image is obsolete. Lines are the one added to /etc/coova/up.sh Should be the 8-10 lines at the end. I dont have the hotspot available for now
-------- Message d'origine --------De : Victor Maldonado <notifications@github.com> Date : 06/01/2018  17:32  (GMT+01:00) À : pihomeserver/Kupiki-Hotspot-Script <Kupiki-Hotspot-Script@noreply.github.com> Cc : Pi Home Server <pihomeserver@gmail.com>, Comment <comment@noreply.github.com> Objet : Re: [pihomeserver/Kupiki-Hotspot-Script] Captive portal without ethernet (#75) 
Hi, i'm using your prepare .img

Would you know what number of lines they are?
I'm not able to make it work. Thanks!

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/pihomeserver/Kupiki-Hotspot-Script","title":"pihomeserver/Kupiki-Hotspot-Script","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/pihomeserver/Kupiki-Hotspot-Script"}},"updates":{"snippets":[{"icon":"PERSON","message":"@Tywors in #75: Hi, i'm using your prepare .img\r\nWould you know what number of lines they are?\r\n\r\nI'm not able to make it work. Thanks!"}],"action":{"name":"View Issue","url":"https://github.com/pihomeserver/Kupiki-Hotspot-Script/issues/75#issuecomment-355758158"}}}

**Comment by acujl** [comment]
Erased this lines
`ipt -I POSTROUTING -t nat -o $HS_WANIF -j MASQUERADE

LOCAL_IP=`ifconfig $HS_WANIF | grep -Eo "inet (addr:)?([0-9]\.){3}[0-9]" | gr$
LOCAL_LAN=`ip -o -f inet addr show | grep $HS_WANIF | awk '/scope global/ {prin$
ipt -I FORWARD 1 -i $TUNTAP -d $LOCAL_LAN -j DROP
ipt -I INPUT 1 -i $TUNTAP -d $LOCAL_IP -j DROP`

Hotspot on, captive portal dead

**Comment by acujl** [comment]
Put them back, and now captive portal is still dead '-'

**Comment by pihomeserver** [comment]
Dis you stop - start the service ? Reboot ? If it still does not work then you sure changed something else before

**Comment by acujl** [comment]
Just reinstalled everything. working.

Now I'm trying to see if this work without Ethernet plugged in

**Comment by acujl** [comment]
Any thoughts on that? @pihomeserver

**Comment by acujl** [comment]
Found this: https://github.com/coova/coova-chilli/issues/212

Any suggestions? Still nothing when I unplug the wlan0

**Comment by pihomeserver** [comment]
Sounds to help in the good direction with the capability to have a landing page in case Internet is not up. If you give a try i'm curious of the result to add it in the script

**Comment by s13884** [comment]
@Tywors  @acujl  Are you able to configure the chilli without internet?

**Comment by Tywors** [comment]
@s13884 yes, without internet yes. But i need connect (raspberry in my case) to a wifi mobile phone without internet. But work :)

**Comment by s13884** [comment]
@Tywors So it requires Ethernet cable plugged in without internet? Or rpi wifi connected to AP?  
Also would you please help me to configure the same?

**Comment by Tywors** [comment]
@s13884 sorry i don't have time to help.. but its simple i only connect rpi to AP without internet.

**Comment by s13884** [comment]
Ok @Tywors, is it possible for you to share the chilli config?

**Comment by doxendine** [comment]
Hi. I have a similar question. I'm trying to setup a captive portal with only a Raspberry Pi. No internet & no router. 
The purpose is (in a no wifi situation) to have a custom portal page to allow users to connect via wifi to the page & download different files on the Raspberry Pi.
Do you know if this is possible via Kupiki or other option?

Thanks for your help.
